### PR TITLE
TypeRelaxed<>::clone_with_new_inputs thread safety fix

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
TypeRelaxed::clone_with_new_inputs had a data race for inputs source output of TypeRelaxed operation which is being cloned.
Original operation was passed to TR ctor which called copy ctor of ov::Node. Copy ctor creates new input objects and to do so it needs to reach input source output [(link)](https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/node.cpp#L46). Additionally, TypeRelaxed::clone_with_new_inputs calls for replace_source_outputs of the node [(link)](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/include/ov_ops/type_relaxed.hpp#L359).
Since this input source output are still the same object we experience data race. 

### Tickets:
 - *79554, 78837*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/subgraph.cpp [hunk 1]
```diff
@@ -89,14 +89,7 @@ void Snippet::copy_snippet() {
         auto new_input = std::make_shared<ngraph::opset1::Parameter>(input.get_element_type(), input.get_partial_shape());
         subgraph_node_inputs.push_back(new_input);
     }
-    std::shared_ptr<ov::Model> new_body = nullptr;
-    // Ticket[79554]: TypeRelaxed ops aren't thread safe so we use mutex to avoid collision in throughput mode
-    if (original_snippet->has_type_relaxed_ops()) {
-        std::lock_guard<std::mutex> lock(*context->getSharedMutex());
```
Comment (ilyachur) [inline src/plugins/intel_cpu/src/nodes/subgraph.cpp:95]: I think we can remove `getSharedMutex()` function I see that we use this method only here.

**Review by ilyachur — COMMENTED** [review]
Could you please add new tests to cover the change?

**Review by a-sidorova — APPROVED** [review]
CPU Part LGTM

**Comment by ilya-lavrenov** [comment]
@jane-intel  could you please port it to LTS? It cherry-picks with merge conflicts..

**Comment by jane-intel** [comment]
The benefits are questionable (only compile_model time gets a bit faster) and there is a risk. I conclude that it isn't worth it. Removing the label

**Pull request merged**